sympy/parsing/sympy_parser.py

```diff
@@ -1000,6 +1051,57 @@ def parse_expr(s, local_dict=None, transformations=standard_transformations,
                 raise TypeError(filldedent('''
                     a transformation should be function that
                     takes 3 arguments'''))
+
+    preprocess = (default + extra).lower()
```

Comment:
What is this extra code? It seems like something that should go in a transformer function.

Reply:
`default` and `extra` both have to do with processing of numbers. It just seems like dealing with them up front is easier than working with the tokenizer. The defaults deal with things that must be dealt with while the extra (exponentials with decimal points which might otherwise parse partially as symbols and complex python j recognition) are easily handled once the numbers have all be identified. These are described with examples more fully in the docstring.

Reply:
I think these should go in the relevant transformer(s). It should be handleable at the token level. You just need to figure out how the relevant strings are tokenized. It looks like they would mainly be tokenized as some combination of NAME, DOT, and NUMBER. 

As we deal with more and more complicated non-Python constructs, we may want to build our own tokenizer instead of trying to work around the Python one.

We could add the ability to pass preprocessors to parse_expr, but I don't think the added complexity is needed for this case. We should be able to split these cases via the transformers themselves.